### DISCUSSION TITLE
Can't read shapefiles unless a bounding box is provided.

### DISCUSSION THREAD

**Discussion by enkicoma** [systemed/tilemaker#454] [Q&A] [answered]
Hi guys,

may I ask for some help please, I am quite new to this and is my first time using `tilemaker`

I am getting an error after:
`./tilemaker --input ../../merged-countries.osm.pbf --output ../../moldova-boxed-tilemaker2.mbtiles  --process ../resources/process-openmaptiles.lua --config ../resources/config-openmaptiles.json`

```js
Layer place (z0-14)
Layer boundary (z0-14)
Layer poi (z12-14)
Layer poi_detail (z14-14) -> poi
Layer housenumber (z14-14)
Layer waterway (z8-14)
Layer waterway_detail (z12-14) -> waterway
Layer transportation (z4-14)
Layer transportation_name (z8-14)
Layer transportation_name_mid (z12-14) -> transportation_name
Layer transportation_name_detail (z14-14) -> transportation_name
Layer building (z13-14)
Layer water (z6-14)
Layer ocean (z0-14) -> water
Layer water_name (z14-14)
Layer water_name_detail (z14-14) -> water_name
Layer aeroway (z11-14)
Layer aerodrome_label (z10-14)
Layer park (z11-14)
Layer landuse (z4-14)
Layer urban_areas (z4-8) -> landuse
Layer landcover (z0-14)
Layer ice_shelf (z0-9) -> landcover
Layer glacier (z2-9) -> landcover
Layer mountain_peak (z11-14)
Can't read shapefiles unless a bounding box is provided.
`

**Comment by systemed** [comment, marked as answer]
I’d guess that whatever tool you’ve used to merge the pbfs hasn’t written a bounding box when doing so. Have a look at the options for that tool and see if there’s one that will force it to do so.

**Reply by enkicoma** [reply]
I did like this, with osmium:
`osmium merge moldova-latest.osm.pbf romania-latest.osm.pbf ukraine-latest.osm.pbf -o merged-countries.osm.pbf`
and after merge::
`osmium extract --bbox=26.49,45.21,30.56,48.57 --set-bounds --strategy=smart merged-countries.osm.pbf   --output moldova-boxed.osm.pbf `

to me looks ok?

**Reply by systemed** [reply]
Looks ok. You can use `osmium fileinfo` to check that the bounds have indeed been set. If you're still having problems then upload the file somewhere and I can take a look if I have a free moment.

**Reply by enkicoma** [reply]
```js
 osmium fileinfo moldova-boxed.osm.pbf
File:
  Name: moldova-boxed.osm.pbf
  Format: PBF
  Compression: none
  Size: 123763369
Header:
  Bounding boxes:
    (26.49,45.21,30.56,48.57)
  With history: no
  Options:
    generator=osmium/1.14.0
    pbf_dense_nodes=true
    pbf_optional_feature_0=Sort.Type_then_ID
    sorting=Type_then_ID
  ```

**Reply by enkicoma** [reply]
very kind and thx for your time, I can't attach it here `we don't support that file type`. Can do through discord or email

**Reply by enkicoma** [reply]
oh wait... I used the wrong file in the input: 
`/tilemaker --input ../../merged-countries.osm.pbf ` needs to be: `moldova-boxed.osm.pbf`

many thanks for the hint

does work:
`Filled the tileset with good things at ../../moldova-boxed-tilemaker2.mbtiles`

**Reply by systemed** [reply]
That's great!
